bluetooth/src/androidLibMain/kotlin/BluetoothMonitor.kt

```diff
@@ -46,26 +49,23 @@ actual class BluetoothMonitor internal constructor(
         }
     }
 
-    private val isPoweredOn: Boolean
-        get() = bluetoothAdapter?.isEnabled == true
-
-    private val _isEnabled = MutableStateFlow(isPoweredOn)
-    actual val isEnabled = _isEnabled.asStateFlow()
+    override val isServiceEnabled: Boolean
+        get() = bluetoothAdapter.isEnabled
 
-    actual fun startMonitoring() {
-        if (bluetoothAdapter == null) return
+    override fun startMonitoring() {
+        super.startMonitoring()
         applicationContext.registerReceiver(
             availabilityBroadcastReceiver,
             IntentFilter(BluetoothAdapter.ACTION_STATE_CHANGED)
         )
     }
 
-    actual fun stopMonitoring() {
-        if (bluetoothAdapter == null) return
+    override fun stopMonitoring() {
+        super.stopMonitoring()
         applicationContext.unregisterReceiver(availabilityBroadcastReceiver)
     }
 
     private fun updateEnabledState() {
-        _isEnabled.value = isPoweredOn
+        _isEnabled.value = isServiceEnabled
```

Comment:
This function now can be removed and inside availabilityBroadcastReceiver call abstract class's one

Reply:
336d436